image/manifest.go

```diff
@@ -238,5 +247,27 @@ loop:
 			return errors.Wrap(err, "error changing time")
 		}
 	}
+
+	return nil
+}
+
+func applyOwnership(path string, hdr *tar.Header) error {
+	switch hdr.Typeflag {
+	case tar.TypeSymlink:
+		if err := os.Lchown(path, hdr.Uid, hdr.Gid); err != nil {
+			return errors.Wrap(err, "error chowing symlink")
+		}
+
+	case tar.TypeReg, tar.TypeRegA:
+		if err := os.Chown(path, hdr.Uid, hdr.Gid); err != nil {
+			return errors.Wrap(err, "error chowing file")
+		}
+
+	case tar.TypeDir:
+		if err := os.Chown(path, hdr.Uid, hdr.Gid); err != nil {
+			return errors.Wrap(err, "error chowing directory")
```

Comment:
Probably not worth splitting cases just for file/directory in the error message.  Can we roll this in with `tar.TypeReg`?  If the `Chown` error doesn't already include it (I haven't checked), it would be nice to include the path.